envoy/network/listener.h

```diff
@@ -107,6 +107,16 @@ class UdpListenerConfig {
 
 using UdpListenerConfigOptRef = OptRef<UdpListenerConfig>;
 
+/**
+ * Configuration for an internal listener.
+ */
+class InternalListenerConfig {
```

Comment:
It looks like this isn't used yet for anything. Delete until it's needed?

Reply:
This is used by the ConnectionHandler to determine if the listener type is Internal.
I feel this is cleaner than find the listen address and query the address type then dispatch by the address type.

In fact I almost make the ConnectionHandler not relying on the address type.